### PR TITLE
#7942 test_setupplan.py migrate from testdir to Pytester

### PR DESCRIPTION
**Pull request by itsmegarvi**

### CODE REVIEW

File: testing/test_setupplan.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
-def test_show_fixtures_and_test(testdir, dummy_yaml_custom_test):
+from _pytest.pytester import Pytester
+def test_show_fixtures_and_test(pytester: Pytester, dummy_yaml_custom_test):
```
Comment (nicoddemus) [inline testing/test_setupplan.py:2]:
```suggestion
def test_show_fixtures_and_test(pytester: Pytester, dummy_yaml_custom_test: None) -> None:
```

File: testing/test_setupplan.py [hunk 2]
```diff
@@ -11,15 +12,15 @@ def test_arg(arg):
...
+    result = pytester.runpytest("--setup-plan")
     assert result.ret == 0
 
     result.stdout.fnmatch_lines(
         ["*SETUP    F arg*", "*test_arg (fixtures used: arg)", "*TEARDOWN F arg*"]
     )
 
 
-def test_show_multi_test_fixture_setup_and_teardown_correctly_simple(testdir):
+def test_show_multi_test_fixture_setup_and_teardown_correctly_simple(pytester: Pytester):
```
Comment (nicoddemus) [inline testing/test_setupplan.py:23]:
```suggestion
def test_show_multi_test_fixture_setup_and_teardown_correctly_simple(pytester: Pytester) -> None:
```

File: testing/test_setupplan.py [hunk 3]
```diff
@@ -66,9 +67,9 @@ def test_two(self, fix):
     assert teardown_count == 1
 
 
-def test_show_multi_test_fixture_setup_and_teardown_same_as_setup_show(testdir):
+def test_show_multi_test_fixture_setup_and_teardown_same_as_setup_show(pytester: Pytester):
```
Comment (nicoddemus) [inline testing/test_setupplan.py:70]:
```suggestion
def test_show_multi_test_fixture_setup_and_teardown_same_as_setup_show(pytester: Pytester) -> None:
```

**Review by nicoddemus — CHANGES_REQUESTED** [review]
Looks good @itsmegarvi, thanks for the contribution!

Left a few comments, besides fixing those, please add yourself to `AUTHORS` as well. 👍

**Comment by itsmegarvi** [comment]
> Looks good @itsmegarvi, thanks for the contribution!
> 
> Left a few comments, besides fixing those, please  add yourself to `AUTHORS` as well. 👍
i will make the changes
how do i add myself to authors ?

**Comment by nicoddemus** [comment]
> how do i add myself to authors ?

Ahh sorry I should have been clearer: there's an `AUTHORS` file in the root of the repository, please add yourself to the list sorted alphabetically.

Also seems like linting has failed, can you please run linting into the code? If you now tox, it is just a matter of running `tox -e linting`, or you can take a look at full instructions in [CONTRIBUTING](https://github.com/pytest-dev/pytest/blob/master/CONTRIBUTING.rst#preparing-pull-requests). 👍

**Review by nicoddemus — APPROVED** [review]
Thanks @itsmegarvi!

**Pull request merged**